sys/net/application_layer/nanocoap/sock.c

```diff
@@ -369,9 +370,14 @@ static int _get_put_cb(void *arg, coap_pkt_t *pkt)
 
 ssize_t nanocoap_sock_get(nanocoap_sock_t *sock, const char *path, void *buf, size_t len)
 {
-    /* buffer for CoAP header */
-    uint8_t buffer[CONFIG_NANOCOAP_BLOCK_HEADER_MAX];
```

Comment:
isn't any of these variants prone to be overflown by a long path? (the path seems to go a long way without being length checked)

Reply:
Yes and `coap_opt_put_uri_pathquery()` has no way to check the size of the buffer, it will happily write beyond it and the caller can then check if an overflow happened (at which point it's already too late).